gcloud-java-dns/src/main/java/com/google/gcloud/dns/DnsBatch.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.dns;
+
+import com.google.gcloud.Page;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * A batch of operations to be submitted to Google Cloud DNS using a single HTTP request.
+ */
+public class DnsBatch {
+
+  private List<Request> requests = new LinkedList<>();
+  private Dns dns;
+
+  /**
+   * An operation to be submitted to Google Cloud DNS within this batch. Only an subset of the class
+   * attributes appropriate for the represented operation is initialized. Refer to the class method
+   * and attribute documentation for the specific fields.
+   */
+  public static class Request {
```

Comment:
Comments from the previous PR still apply:

- I would still consider making this `Serializable` and implement `equals` and `hashCode`
- I don't remember what we decided with respect to having a class hierarchy here. Did we get some feedback from @ajkannan and @aozarov